cornell-dti/office-hours
# Exporting calendar redesign

### Summary

Implements Lucy's design changes for the exporting Google/Apple calendar feature.

Originals:

![image](https://github.com/user-attachments/assets/377c198c-4cfa-46e6-a863-2a582a6c0e9e)
![image](https://github.com/user-attachments/assets/5ea0241c-f9fa-4cc8-8b88-5ff6ff393cf1)

Figma redesigns:

![image](https://github.com/user-attachments/assets/91a5d036-40c3-4404-9be5-50792e392bbd)
![image](https://github.com/user-attachments/assets/0f4b52e5-a527-4335-b359-94ede1839943)
![image](https://github.com/user-attachments/assets/9a5c6a01-98e6-492e-a27e-549f074ea106)

### Actual Changes

- Got rid of bookmark
- Changed TA unfinished indicator to be a red dot when there is someone in "Ahead"

![image](https://github.com/cornell-dti/office-hours/assets/57076607/4d88b754-cf3b-491a-8af6-cd7afa47ef95)

![image](https://github.com/cornell-dti/office-hours/assets/57076607/efcfec9e-b907-4a11-9bba-1566b38ed19b)

Redesigned modal:

- Buttons are now vertical
- Functionality is the same
- Got rid of repeating office hour title, now is just course code and OH name
- Added date and time as subtitle

![image](https://github.com/cornell-dti/office-hours/assets/57076607/a0b9432e-fe3f-43f8-803c-c69065cb2c10)

With overflowing text:

![image](https://github.com/cornell-dti/office-hours/assets/57076607/b4473cb4-0e7d-4969-8681-f42367b8ac1e)


### Test Plan

- Change the "Ahead" number of students and make sure red dot appears when there is at least one, otherwise should go away
- Make sure Calendar Export modal shows expected name, date, and times for any office hour
- Check if styling remains consistent for long OH names
- Check if buttons work as expected

### Notes

- Added dependency (ran yarn add @mui/icons-material)

### Breaking Changes 

None

### Checklist
